src/zarr/registry.py

```diff
@@ -5,11 +5,15 @@
 from importlib.metadata import entry_points as get_entry_points
 from typing import TYPE_CHECKING, Any, Generic, TypeVar
 
+import numpy as np
```

Comment:
The changes in this module are a bit hacky. They are mainly just to show how something might work - the actual details can certainly be changed quite a bit.